app/src/main/java/com/distillery/interview/ui/MainActivity.kt

```diff
@@ -1,20 +1,40 @@
 package com.distillery.interview.ui
 
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
 import com.distillery.interview.R
 import com.distillery.interview.ui.viewpager.WeatherCollectionFragment
 
-class MainActivity : AppCompatActivity(){
+class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        if(savedInstanceState == null) {
+        if (savedInstanceState == null) {
             supportFragmentManager.beginTransaction()
                 .replace(R.id.root, WeatherCollectionFragment())
                 .commitAllowingStateLoss()
         }
+
+        requestLocationPermissions()
+    }
+
+    private fun requestLocationPermissions() {
+        try {
+            if (ActivityCompat.checkSelfPermission(this,
+                    android.Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED
+            ) {
+                ActivityCompat.requestPermissions(this,
```

Comment:
If we ask for a permission we need to make sure we got the OK from the user, so, you must override `onRequestPermissionsResult`.

Reply:
yes, I added it later but I can better do it here